nixos/modules/security/dhparams.nix

```diff
@@ -0,0 +1,56 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.security.dhparams;
+  dir = "/var/cache/dhparams";
+in
+{
+  options = {
+    security.dhparams = mkOption {
+      description =
+        ''
+          Diffie-Hellman parameters to generate.
+
+          The value is the size (in bits) of the DH params to generate. The
+          generated DH params path can be found in `${dir}/[name].pem`
```

Comment:
Hmm, I get the issue but can't think of a clean UI to do it. Adding a `.path` requires to move the size to a .size argument (that's OK), but then people could override the `.path` argument… which would not be an issue, if I had any idea of how to make a default value for an argument dependent on the name of its parent (to be able to default to `${dir}/[name].pem`.

Do you have an idea? Or would just putting a `security.dhparamsPath` defaulting to `/var/cache/dhparams` be OK with your use case?